addons/web_editor/controllers/main.py

```diff
@@ -318,7 +324,18 @@ def _attachment_create(self, name='', data=False, url=False, res_id=False, res_m
         else:
             raise UserError(_("You need to specify either data or url to create an attachment."))
 
-        attachment = request.env['ir.attachment'].create(attachment_data)
+        # If the user is a portal and has access rights on the res_model (and on the res_id if it exists), we can create
+        # the attachment. Otherwise, one of the check will raise an error and the attachment will not be created
+        if request.env.user.has_group('base.group_portal'):
+            request.env[res_model].check_access_rights('write')
+            if res_id:
+                request.env[res_model].browse(res_id).check_access_rule('write')
+            attachment = request.env['ir.attachment'].sudo().create(attachment_data)
+            if not attachment_data['public']:
+                attachment.sudo().generate_access_token()
```

Comment:
Could you explain why is this needed? (Or most exactly, why wasn't it needed before this PR?)

Reply:
Maybe this was answered elsewhere, but if I recall correctly this is to short-circuit in this use this code:

https://github.com/odoo/odoo/blob/26fecc13c16c8028d6d1aafc62ef6f3c0d00643f/addons/web_editor/static/src/js/wysiwyg/widgets/media.js#L441-L449

that code works, but not in the use case we are solving (someone that does not have necessary access right).

So this code is doing the same thing in sudo but only at the creation of the attachment.

Reply:
I was still a bit skeptical about this (couldn't figure exactly why you needed to do that) and didn't read your comment.
The reason is indeed that after the attachment has been created by the editor RPC, it sends another RPC to generate the missing access_token if the attachment is not public. This RPC will fail in the case of a portal user.